src/nspawn/nspawn-register.c

```diff
@@ -108,6 +108,34 @@ static int append_controller_property(sd_bus *bus, sd_bus_message *m) {
         return 0;
 }
 
+static int can_set_coredump_receive(sd_bus *bus) {
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_free_ char *path = NULL;
+        int b, r;
+
+        assert(bus);
+
+        path = unit_dbus_path_from_name(SPECIAL_INIT_SCOPE);
+
+        r = sd_bus_get_property_trivial(
+                        bus,
+                        "org.freedesktop.systemd1",
+                        path,
+                        "org.freedesktop.systemd1.Scope",
+                        "CoredumpReceive",
+                        &error,
+                        'b', &b);
+        if (r < 0) {
+                if (!sd_bus_error_has_names(&error, SD_BUS_ERROR_UNKNOWN_PROPERTY, SD_BUS_ERROR_PROPERTY_READ_ONLY))
+                        log_debug_errno(r, "Failed to determine if CoredumpReceive can be set, ignoring: %s",
+                                        bus_error_message(&error, r));
+
+                return false;
+        }
+
+        return true;
```

Comment:
This block can be simplified by returning `r >= 0` at the end and only putting the log statement beneath the `if`